src/index.ts

```diff
@@ -58,16 +76,26 @@ class ImageKit {
    *
    * @param uploadOptions
    */
-  upload(
-    uploadOptions: UploadOptions,
-    callback?: (err: Error | null, response: UploadResponse | null) => void,
-    options?: Partial<ImageKitOptions>,
-  ): void {
+  upload(uploadOptions: UploadOptions, options?: Partial<ImageKitOptions>): Promise<IKResponse<UploadResponse>>
+  upload(uploadOptions: UploadOptions, callback: (err: Error | null, response: IKResponse<UploadResponse> | null) => void, options?: Partial<ImageKitOptions>): void;
+  upload(uploadOptions: UploadOptions, callbackOrOptions?: ((err: Error | null, response: IKResponse<UploadResponse> | null) => void) | Partial<ImageKitOptions>, options?: Partial<ImageKitOptions>): void | Promise<IKResponse<UploadResponse>> {
+    let callback;
+    if (typeof uploadOptions !== "object") {
+      throw ("First parameter needs to be object");
```

Comment:
Change to `uploadOptions should be an object`

Reply:
Actually, the user doesn't know that the first option is called uploadOptions.

Reply:
done